implementations/micrometer-registry-dynatrace/src/main/java/io/micrometer/dynatrace/DynatraceMeterRegistry.java

```diff
@@ -231,21 +235,75 @@ void putCustomMetric(final DynatraceMetricDefinition customMetric) {
 
     private void postCustomMetricValues(String type, List<DynatraceTimeSeries> timeSeries, String customDeviceMetricEndpoint) {
         try {
-            httpClient.post(customDeviceMetricEndpoint)
-                    .withJsonContent("{\"type\":\"" + type + "\"" +
-                            ",\"series\":[" +
-                            timeSeries.stream()
-                                    .map(DynatraceTimeSeries::asJson)
-                                    .collect(joining(",")) +
-                            "]}")
-                    .send()
-                    .onSuccess(response -> logger.debug("successfully sent {} metrics to Dynatrace.", timeSeries.size()))
-                    .onError(response -> logger.error("failed to send metrics to dynatrace: {}", response.body()));
+            for (Tuple<String, Integer> postMessage : createPostMessages(type, timeSeries)) {
+                httpClient.post(customDeviceMetricEndpoint)
+                        .withJsonContent(postMessage.x)
+                        .send()
+                        .onSuccess(response -> logger.debug("successfully sent {} metrics to Dynatrace ({} bytes).",
+                                postMessage.y, postMessage.x.getBytes(UTF_8).length))
+                        .onError(response -> logger.error("failed to send metrics to dynatrace: {}", response.body()));
+            }
         } catch (Throwable e) {
             logger.error("failed to send metrics to dynatrace", e);
         }
     }
 
+    // VisibleForTesting
+    List<Tuple<String, Integer>> createPostMessages(String type, List<DynatraceTimeSeries> timeSeries) {
+        final StringBuilder sb = new StringBuilder(1024);
```

Comment:
I don't think there is any benefit to using a `StringBuilder` here. The Java compiler will optimize String concatenation in simple cases like this ("string" + a + "string" + b) to use a `StringBuilder` in the bytecode. I think only in the case of concatenating in a loop is an explicit `StringBuilder` needed because it isn't optimized automatically.

Reply:
new StringBuilder() uses a 16 char array so initialising the stringbuilder with a bigger array avoids the runtime from having to create new arrays as the string gets longer

```
   public StringBuilder() {
        super(16);
    }
```